test/MinimalDelegation.execute.invariant.t.sol

```diff
@@ -0,0 +1,245 @@
+            console2.logBytes(debugCalldata);
+            console2.logBytes(revertData);
+            revert("uncaught revert");
+        }
+    }
+}
+
+/// @title MinimalDelegationExecuteInvariantTest
+contract MinimalDelegationExecuteInvariantTest is TokenHandler, DelegationHandler {
+    using KeyLib for Key;
+    using TestKeyManager for TestKey;
+    using CallUtils for Call;
+    using CallUtils for Call[];
+    using WrappedDataHash for bytes32;
+
+    MinimalDelegationExecuteInvariantHandler internal invariantHandler;
+
+    address public untrustedCaller = makeAddr("untrustedCaller");
+    uint256 public untrustedPrivateKey = 0xdead;
+
+    address internal sender = makeAddr("sender");
+
+    TestKey[] internal _keys;
+    TestKey internal untrustedKey;
+
+    function setUp() public {
+        setUpDelegation();
+        setUpTokens();
+
+        vm.deal(address(signerAccount), 100e18);
+        tokenA.mint(address(signerAccount), 100e18);
+        tokenB.mint(address(signerAccount), 100e18);
+
+        address[] memory callers = new address[](2);
+        // Add trusted root caller
+        callers[0] = address(signerAccount);
+        vm.label(callers[0], "signerAccount");
+        // Add untrusted caller
+        callers[1] = untrustedCaller;
+        vm.label(callers[1], "untrustedCaller");
+
+        untrustedKey = TestKeyManager.withSeed(KeyType.Secp256k1, untrustedPrivateKey);
+
+        // Add trusted root key
+        _keys.push(TestKeyManager.withSeed(KeyType.Secp256k1, signerPrivateKey));
+        // Add untrusted key
+        _keys.push(untrustedKey);
+
+        SetupParams memory params = SetupParams({
+            _signerAccount: signerAccount,
+            _keys: _keys,
+            _callers: callers,
+            _tokenA: address(tokenA),
+            _tokenB: address(tokenB)
+        });
+        invariantHandler = new MinimalDelegationExecuteInvariantHandler(params);
+
+        // Explicitly target the wrapped execute functions in the handler
+        bytes4[] memory selectors = new bytes4[](2);
+        selectors[0] = MinimalDelegationExecuteInvariantHandler.executeBatchedCall.selector;
+        selectors[1] = MinimalDelegationExecuteInvariantHandler.executeWithOpData.selector;
+        FuzzSelector memory selector = FuzzSelector({addr: address(invariantHandler), selectors: selectors});
+
+        targetSelector(selector);
+        targetContract(address(invariantHandler));
+
+        // Sender is the address used to call the invariantHandler, not important
+        targetSender(sender);
+    }
+
+    /// @notice Verifies that the root key can always revoke other keys
+    function invariant_rootKeyCanAlwaysRevokeOtherSigningKeys() public {
+        // Ensure key exists
+        bytes32 keyHash = untrustedKey.toKeyHash();
+        try signerAccount.getKey(keyHash) {
+            vm.prank(address(signerAccount));
+            signerAccount.revoke(keyHash);
```

Comment:
shouldn't this assert that the revoke has actually removed the state from the contract and/or another call from that key is invalid?

Reply:
also how would we know if the state ever catches the first part of the try? ie that the specific untrusted key hash ever existed on the contract?

Reply:
agree that it should assert it removed the key from the contract state but the `assert` in the catch branch checks that the only reason it should revert is if the key does not exist (it was probably removed by the fuzzed invariant test)

Reply:
removed these because `untrustedSigningKey` will never be registered on the account now